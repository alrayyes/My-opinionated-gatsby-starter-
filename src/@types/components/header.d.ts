export interface HeaderInterface {
  siteTitle: string
}
