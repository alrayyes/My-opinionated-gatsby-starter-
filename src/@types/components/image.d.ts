import { FluidObject } from "gatsby-image"

export interface ImageDataInterface {
  placeholderImage: {
    childImageSharp: {
      fluid: FluidObject
    }
  }
}
