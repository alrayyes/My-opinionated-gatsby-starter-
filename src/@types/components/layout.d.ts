export interface LayoutInterface {
  children: React.ReactNode
}

export interface LayoutDataInterface {
  site: {
    siteMetadata: {
      title: string
    }
  }
}
