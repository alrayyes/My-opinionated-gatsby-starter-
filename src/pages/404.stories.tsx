import { storiesOf } from "@storybook/react"
import React from "react"
import NotFoundPage from "./404"

storiesOf("404 page", module).add(`default`, () => <NotFoundPage />)
