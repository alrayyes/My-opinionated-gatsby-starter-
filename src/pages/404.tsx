import React from "react"

import Layout from "../components/layout"
import SEO from "../components/seo"
import styled from "@emotion/styled"
import tw from "twin.macro"

const Header = styled.h1`
  ${tw`text-5xl font-bold`}
`

const Text = styled.p`
  ${tw`mt-6 text-xl`}
`

const NotFoundPage: React.FC = () => (
  <Layout>
    <SEO title="404: Not found" />
    <Header>NOT FOUND</Header>
    <Text>You just hit a route that doesn&#39;t exist... the sadness.</Text>
  </Layout>
)

export default NotFoundPage
