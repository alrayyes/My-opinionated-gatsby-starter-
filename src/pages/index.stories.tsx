import { storiesOf } from "@storybook/react"
import React from "react"
import IndexPage from "./index"

storiesOf("Index page", module).add(`default`, () => <IndexPage />)
