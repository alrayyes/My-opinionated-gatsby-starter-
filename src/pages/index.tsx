import React from "react"

import Layout from "../components/layout"
import Image from "../components/image"
import SEO from "../components/seo"
import styled from "@emotion/styled"
import tw from "twin.macro"
import { Link } from "gatsby"

const IndexPage: React.FC = () => {
  const ImageContainer = styled.div`
    ${tw`max-w-xs mb-6 mt-4`}
  `

  return (
    <Layout>
      <SEO title="Home" />
      <h1>Hi people</h1>
      <p>Welcome to your new Gatsby site.</p>
      <p>Now go build something great.</p>
      <ImageContainer>
        <Image />
      </ImageContainer>
      <Link to="/page-2/">Go to page 2</Link> <br />
      <Link to="/using-typescript/">Go to &quot;Using TypeScript&quot;</Link>
    </Layout>
  )
}

export default IndexPage
