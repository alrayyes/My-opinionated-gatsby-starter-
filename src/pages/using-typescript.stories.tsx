import { storiesOf } from "@storybook/react"
import React from "react"
import UsingTypescript from "./using-typescript"
import { UsingTypescriptDataInterface } from "../@types/pages/using-typescript"

const data: UsingTypescriptDataInterface = {
  data: {
    site: {
      buildTime: "2020-06-05T10:23:36.000Z",
    },
  },
}
storiesOf("Using Typescript page", module).add(`default`, () => (
  <UsingTypescript data={data.data} path={"/using-typescript/"} />
))
