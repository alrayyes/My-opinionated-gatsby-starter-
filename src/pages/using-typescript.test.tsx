import React from "react"
import renderer from "react-test-renderer"
import UsingTypescript from "./using-typescript"
import { UsingTypescriptDataInterface } from "../@types/pages/using-typescript"
import { SEODataInterface } from "../@types/components/seo"
import * as Gatsby from "gatsby"

const useStaticQuery = jest.spyOn(Gatsby, "useStaticQuery")

describe("NotFoundPage", () => {
  beforeEach(() => {
    const data: SEODataInterface = {
      site: {
        siteMetadata: {
          title: "Gatsby Default Starter",
          description:
            "Kick off your next, great Gatsby project with this default starter. This barebones starter ships with the main Gatsby configuration files you might need.",
          author: "@gatsbyjs",
        },
      },
    }

    useStaticQuery.mockImplementation(() => data)
  })

  it("renders correctly", () => {
    const data: UsingTypescriptDataInterface = {
      data: {
        site: {
          buildTime: "2020-06-04T09:48:20.000Z",
        },
      },
    }

    expect(
      renderer
        .create(<UsingTypescript data={data.data} path={"/path"} />)
        .toJSON()
    ).toMatchSnapshot()
  })
})
