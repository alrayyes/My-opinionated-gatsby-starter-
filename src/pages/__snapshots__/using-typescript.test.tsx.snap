// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotFoundPage renders correctly 1`] = `
Array [
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-1pcnel4-Header{background:rebeccapurple;margin-bottom:1.45rem;}",
      }
    }
    data-emotion-css="1pcnel4-Header"
  />,
  <header
    className="css-1pcnel4-Header e1qr1m9j0"
  >
    <style
      dangerouslySetInnerHTML={
        Object {
          "__html": ".css-pf9rd5-Container{margin:0 auto;max-width:960;padding:1.45rem 1.0875rem;}",
        }
      }
      data-emotion-css="pf9rd5-Container"
    />
    <div
      className="css-pf9rd5-Container e1qr1m9j1"
    >
      <style
        dangerouslySetInnerHTML={
          Object {
            "__html": ".css-1ikcnub-H1{margin:0;}",
          }
        }
        data-emotion-css="1ikcnub-H1"
      />
      <h1
        className="css-1ikcnub-H1 e1qr1m9j2"
      >
        <style
          dangerouslySetInnerHTML={
            Object {
              "__html": ".css-rkbe0q-StyledLink{color:white;-webkit-text-decoration:none;text-decoration:none;}",
            }
          }
          data-emotion-css="rkbe0q-StyledLink"
        />
        <a
          className="css-rkbe0q-StyledLink e1qr1m9j3"
          href="/"
        >
          Gatsby Default Starter
        </a>
      </h1>
    </div>
  </header>,
  <style
    dangerouslySetInnerHTML={
      Object {
        "__html": ".css-15zhpu-Container{margin:0 auto;max-width:960;padding:0 1.0875rem 1.45rem;}",
      }
    }
    data-emotion-css="15zhpu-Container"
  />,
  <div
    className="css-15zhpu-Container e9jw82o0"
  >
    <main>
      <h1>
        Gatsby supports TypeScript by default!
      </h1>
      <p>
        This means that you can create and write 
        <em>
          .ts/.tsx
        </em>
         files for your pages, components etc. Please note that the 
        <em>
          gatsby-*.js
        </em>
         files (like gatsby-node.js) currently don't support TypeScript yet.
      </p>
      <p>
        For type checking you'll want to install 
        <em>
          typescript
        </em>
         via npm and run 
        <em>
          tsc --init
        </em>
         to create a 
        <em>
          .tsconfig
        </em>
         file.
      </p>
      <p>
        You're currently on the page "
        /path
        " which was built on
         
        2020-06-04T09:48:20.000Z
        .
      </p>
      <p>
        To learn more, head over to our
         
        <a
          href="https://www.gatsbyjs.org/docs/typescript/"
        >
          documentation about TypeScript
        </a>
        .
      </p>
      <a
        href="/"
      >
        Go back to the homepage
      </a>
    </main>
    <footer>
      Â© 
      2020
      , Built with
       
      <a
        href="https://www.gatsbyjs.org"
      >
        Gatsby
      </a>
    </footer>
  </div>,
]
`;
